fits2caom2/src/main/java/ca/nrc/cadc/caom2/fits/wcs/Wcs.java

```diff
@@ -379,57 +379,110 @@ public static CoordFunction2D getCoordFunction2D(String utype, FitsMapping mappi
-            {
-                try
-                {
-                    Double cd1 = Double.valueOf(cdelt1);
-                    Double cd2 = Double.valueOf(cdelt2);
-                    Double rotation = crota == null ? 0.0 : Double.valueOf(crota);
-                    Double sign = crotsign == null ? 1.0 : Double.valueOf(crotsign);
-                    
-                    cd11 = cd1 * Math.cos(rotation) * sign;
-                    cd12 = -cd2 * Math.sin(rotation) * sign;
-                    cd21 = cd1 * Math.sin(rotation) * sign;
-                    cd22 = cd2 * Math.cos(rotation) * sign;
-                }
-                catch (NumberFormatException e) { }
+        // Try the PC matrix next.
+        int posAxis1 = mapping.positionAxis1;
+        int posAxis2 = mapping.positionAxis2;
+        String cdelt1 = mapping.getKeywordValue("CDELT" + posAxis1);
+        String cdelt2 = mapping.getKeywordValue("CDELT" + posAxis2);
+
+        if (cdelt1 != null && cdelt2 != null) {
+            Double pc11 = null;
+            Double pc12 = null;
+            Double pc21 = null;
+            Double pc22 = null;
+
+            String kw11 = String.format("PC%d_%d", posAxis1, posAxis1);
+            String s11 = mapping.getKeywordValue(kw11);
+            if (s11 == null) {
+                kw11 = String.format("PC0%d_0%d", posAxis1, posAxis1);
+                s11 = mapping.getKeywordValue(kw11);
+            }
+            if (s11 != null) {
+                pc11 = Double.parseDouble(s11);
+            }
+
+            String kw12 = String.format("PC%d_%d", posAxis1, posAxis2);
+            String s12 = mapping.getKeywordValue(kw12);
+            if (s12 == null) {
+                kw12 = String.format("PC0%d_0%d", posAxis1, posAxis2);
+                s12 = mapping.getKeywordValue(kw12);
+            }
+            if (s12 != null) {
+                pc12 = Double.parseDouble(s12);
+            }
+
+            String kw21 = String.format("PC%d_%d", posAxis2, posAxis1);
+            String s21 = mapping.getKeywordValue(kw21);
+            if (s21 == null) {
+                kw21 = String.format("PC0%d_0%d", posAxis2, posAxis1);
+                s21 = mapping.getKeywordValue(kw21);
+            }
+            if (s21 != null) {
+                pc21 = Double.parseDouble(s21);
+            }
+
+            String kw22 = String.format("PC%d_%d", posAxis2, posAxis2);
+            String s22 = mapping.getKeywordValue(kw22);
+            if (s22 == null) {
+                kw22 = String.format("PC0%d_0%d", posAxis2, posAxis2);
+                s22 = mapping.getKeywordValue(kw22);
+            }
+            if (s22 != null) {
+                pc22 = Double.parseDouble(s22);
+            }
+
+            if (pc11 != null && pc22 != null && pc12 == null && pc21 == null) {
+                pc12 = 0.0;
+                pc21 = 0.0;
+            }
+
+            if (pc11 != null && pc12 != null && pc21 != null && pc22 != null) {
```

Comment:
From my reading of FITS WCS paper 1 this is not quite right: the top of p 1064 says that 

CDi_j = Si * Mij

where S is the scale (CDELT) and Mij are the matrix (PC) values.